mirgecom/__init__.py

```diff
@@ -19,3 +19,43 @@
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 """
+
+import os
+import getpass
+import sys
+import tempfile
+from warnings import warn
+
+from mpi4py import MPI
+
+# This code avoids slow startups due to file system locking when running with
+# large numbers of ranks. See https://github.com/illinois-ceesd/planning/issues/27
+# for details.
+
+size = MPI.COMM_WORLD.Get_size()
+rank = MPI.COMM_WORLD.Get_rank()
+
+if size >= 256 and rank == 0:
+    if "XDG_CACHE_HOME" not in os.environ:
+        skip = False
+
+        for mod in ["pyopencl", "pytools", "loopy"]:
+            if mod in sys.modules:
+                warn(f"{mod} already loaded, not adjusting XDG_CACHE_HOME. Please "
+                  "see https://github.com/illinois-ceesd/planning/issues/27 for "
+                  "information on how to avoid file system overheads by setting "
+                  "the XDG_CACHE_HOME variable in your job scripts.")
+                skip = True
+                break
+
+        if not skip:
+            username = getpass.getuser()
+            dir_prefix = f"/tmp/{username}/"
+            os.makedirs(dir_prefix, exist_ok=True)
```

Comment:
This is still predictable, and I think it can still be used for arbitrary file creation across accounts (by planting a symlink) in `/tmp`. Plus `/tmp` isn't always the directory for temporary files. Just leave as much as possible of this to `tempfile`.

Reply:
Instead of attempting to work around this issue here, would it be enough to just warn the user and point them how to fix it themselves?

Reply:
Implemented the warning only.